pages/content/posts/cycletracker.md

```diff
@@ -0,0 +1,31 @@
+* Data collection and sharing: Data may be shared with advertisers, data brokers, and law enforcement.
+* Data security: Databases run the risk of data breaches.
+
+This is especially concerning in areas that try to regulate or even criminalize reproductive health care, including contraception and abortion. The Florida High School Athletic Association considered a policy to require girls to provide information about their menstrual cycle. Fortunately, this policy was not implemented. Restricting access to abortion, and even access to contraception is not limited to the USA. There is a legitimate concern that MCT data can be used to prevent girls from playing sports, prosecute people who have abortions, and legislate the bodies and pregnant people.
+
+If you are concerned about the privacy risks associated with MCTs, you can create and host your own! There is a step-by-step [introductory-level tutorial to create a menstrual tracking application](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Tutorials/CycleTracker). CycleTracker is a progressive web app (PWA) with a form for the user enters the start and end date of each period and the entered data is stored in the browser’s localStorage. The PWA can be stored on your device and is fully functional when offline. If you follow the steps in the tutorial, you can create the PWA yourself, and host it locally without ever putting any code online. You can install it on your own device. It works when you're offline or when your local server isn’t running, and will continue running even if you delete the original source code. No form data is submitted to a server. As all the data is stored in localStorage over which you have full control. to delete the data, Because you are coding this PWA yourself, you can add additional features, including features to increase privacy. While the tutorial uses the name "CycleTracker", you can change the name to obfuscate the purpose of the app. For example, you could name the app "Manicures" and change the labeling to make it appear that the start date is when you got the manicure and the end date is when your nails needed a re-polish or a press-on nail needed a refill. You're in control of the PWA you develop and the data you enter into it.
+
+If you want to learn how to create PWAs, including basic markup, testing locally and creating a [secure connection](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Tutorials/CycleTracker/Secure_connection), the components of manifest, and iconography file including defining the assets to cache and iconography, and creating the [service worker](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Tutorials/CycleTracker/Service_workers) to implement the cache and enable working offline, work the [CycleTracker tutorial](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Tutorials/CycleTracker).
```

Comment:
```suggestion
If you want to learn how to create PWAs, check out the [CycleTracker tutorial](https://developer.mozilla.org/docs/Web/Progressive_web_apps/Tutorials/CycleTracker). It will take you through all the required steps, including building the basic markup and styles, adding the JavaScript logic, testing locally, and creating a secure connection. It will also give you an opportunity to learn about the components of a PWA manifest file, its iconography, how to define the assets to cache, and how to create a service worker to enable working offline.
```